### PR TITLE
Fixed filter for status in task table

### PR DESCRIPTION
**Pull request by shuqi7**
- PRs https://github.com/apache/incubator-druid/pull/7448 and https://github.com/apache/incubator-druid/pull/7460 caused a regression bug, in which the filter for status in the tasks table does not work because the custom filter in https://github.com/apache/incubator-druid/pull/7448 assumes a primitive value would be provided, while in https://github.com/apache/incubator-druid/pull/7460 the status row provides an object
- This PR fixed the filter by providing a custom filter for the status column

### CODE REVIEW

**Comment by vogievetsky** [comment]
Wow, this is like when two cars merge into the same lane at the same time from right and left

File: web-console/src/utils/general.tsx [hunk 1]
```diff
@@ -66,13 +66,11 @@ export function makeBooleanFilter(): FilterRender {
   };
 }
 
-export function customTableFilter(filter: Filter, row?: any): boolean | string {
-  const id = filter.pivotId || filter.id;
-  const clientSideFilter: boolean = row !== undefined;
-  if (clientSideFilter && row[id] === undefined) {
+export function customTableFilter(filter: Filter, clientSideFilter: boolean, value?: any): boolean | string {
```
Comment (vogievetsky) [inline web-console/src/utils/general.tsx]: This function has no clear meaning, could you please refactor this into two functions one for `clientSideFilter` `true` and one for `false`

File: web-console/src/utils/general.tsx [hunk 2]
```diff
@@ -66,28 +66,48 @@ export function makeBooleanFilter(): FilterRender {
...
+  const needleAndMode: NeedleAndMode = getNeedleAndMode(filter);
+  const needle = needleAndMode.needle;
+  if (needleAndMode.mode === 'exact') {
+    return needle === haystack;
   }
-  if (clientSideFilter) {
-    return targetValue.includes(filterValue);
-  } else {
-    return `${targetValue} LIKE '%${filterValue}%'`;
+  return haystack.includes(needle);
```
Comment (vogievetsky) [inline web-console/src/utils/general.tsx]: includes or startsWith?

File: web-console/src/utils/general.tsx [hunk 3]
```diff
@@ -66,28 +66,48 @@ export function makeBooleanFilter(): FilterRender {
...
+}
+
+export function sqlQueryCustomTableFilter(filter: Filter): string {
+  const columnName = JSON.stringify(filter.id).toLowerCase();
+  const needleAndMode: NeedleAndMode = getNeedleAndMode(filter);
+  const needle = needleAndMode.needle;
+  if (needleAndMode.mode === 'exact') {
+    return `${columnName} = '${needle}'`;
   }
+  return `${columnName} LIKE '%${needle}%'`;
```
Comment (vogievetsky) [inline web-console/src/utils/general.tsx]: I think `${columnName} LIKE '${needle}%` right?

Comment (shuqi7) [reply web-console/src/utils/general.tsx]: I think the original code is indeed `include` not `prefix`. Maybe I should change the mode from `prefix` to `include`, but do we want the prefix or include for the filter?

Comment (vogievetsky) [reply web-console/src/utils/general.tsx]: we want prefix.

**Comment by vogievetsky** [comment]
Looks great, thank you 👍

**Pull request merged**